### PR TITLE
Add channel close correlation

### LINKED ISSUE #868: MsgChannelCloseInit
Right now `MsgChannelCloseInit` messages are unhandled in the relayer.

Do we observe it in the tendermint events? Do we need to handle it?

Implementing a `MsgChannelCloseInit`/`MsgChannelCloseConfirm` channel close flow would look something like this:
https://github.com/cosmos/relayer/blob/91d50f8cd420eff208f063f8695e2636328953f7/relayer/processor/path_processor_internal.go#L363-L405
https://github.com/cosmos/relayer/blob/91d50f8cd420eff208f063f8695e2636328953f7/relayer/processor/path_processor_internal.go#L601-L614
https://github.com/cosmos/relayer/blob/91d50f8cd420eff208f063f8695e2636328953f7/relayer/processor/path_processor_internal.go#L668-L670

This code was written while @jtieri and I were working on channel closures until discovering that `MsgChannelCloseInit` plays no part in the channel close flow involving `MsgTimeout` on an ordered channel.

### PR DESCRIPTION
**Pull request by agouin**
Reintroduce channel close correlation. We had disabled it due to the discoveries needed by #868 

This un-ties the correlation between ordered timeouts and channel closures by simulating an observed `channel_close_init` when the timeout occurs. This means it will work if the `channel_close_init` event is emitted from the timeout message in the transaction events or not. 

`rly tx channel-close` will now flush any timeouts and then finish closing the channel on the other chain with the close-confirm

Resolves #868

### CODE REVIEW

File: interchaintest/ica_channel_close_test.go [hunk 1]
```diff
@@ -0,0 +1,315 @@
...
+		"-y",
+	}
+	_, _, err = chain1.Exec(ctx, registerICA, nil)
+	require.NoError(t, err)
+
+	// Start the relayer and set the cleanup function.
+	err = r.StartRelayer(ctx, eRep, pathName)
+	require.NoError(t, err)
+
+	t.Cleanup(
```
Comment (boojamya) [inline interchaintest/ica_channel_close_test.go:143]: Why is this necessary?

File: interchaintest/ica_channel_close_test.go [hunk 2]
```diff
@@ -0,0 +1,315 @@
...
+		func() {
+			err := r.StopRelayer(ctx, eRep)
+			if err != nil {
+				t.Logf("an error occured while stopping the relayer: %s", err)
+			}
+		},
+	)
```
Comment (boojamya) [inline interchaintest/ica_channel_close_test.go:150]: Why is this necessary?

File: relayer/processor/path_processor_internal.go [hunk 3]
```diff
@@ -488,6 +473,76 @@ func (pp *PathProcessor) unrelayedChannelHandshakeMessages(
...
+		pathEndChannelCloseMessages.Src.messageCache.ChannelHandshake.DeleteMessages(toDeleteSrc)
+		pathEndChannelCloseMessages.Dst.messageCache.ChannelHandshake.DeleteMessages(toDeleteDst)
+		pathEndChannelCloseMessages.Src.channelProcessing.deleteMessages(toDeleteSrc)
+		pathEndChannelCloseMessages.Dst.channelProcessing.deleteMessages(toDeleteDst)
+		toDeleteSrc = make(map[string][]ChannelKey)
+		toDeleteDst = make(map[string][]ChannelKey)
+	}
+
+	for chanKey := range pathEndChannelCloseMessages.DstMsgChannelCloseConfirm {
+		// found open confirm, channel handshake complete. remove all retention
```
Comment (jtieri) [inline relayer/processor/path_processor_internal.go]: this comment seems out of place

**Review by jtieri — APPROVED** [review]
looks good!
